examples/etl_example.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+"""
+Example of using Celery canvas features to build an ETL pipeline.
+This example demonstrates how to chain and group tasks for data processing.
+
+This ETL pipeline example shows how to use Celery's powerful canvas features
+to build a data processing workflow that:
+1. Extracts data from multiple sources in parallel
+2. Merges the extracted data
+3. Transforms the data with computed fields and filtering
+4. Loads the results into a JSON file
+
+Key Celery Features Demonstrated:
+- Parallel task execution using `group`
+- Task chaining using the `|` operator
+- Result passing between tasks
+- Error handling and logging
+
+Pipeline Flow:
+    [extract(source_a), extract(source_b), extract(source_c)] (in parallel)
+                           ↓
+                     merge_data
+                           ↓
+                      transform
+                           ↓
+                        load
+
+To run this example:
+
+1. Start Redis (broker and result backend):
+   ```bash
+   redis-server
+   ```
+
+2. Start a Celery worker:
+   ```bash
+   celery -A examples.etl_example worker --loglevel=INFO
+   ```
+
+3. Run the example:
+   ```bash
+   python examples/etl_example.py
+   ```
+
+The pipeline will extract sample data from three sources in parallel,
+merge the results, apply transformations (adding computed fields and filtering),
+and save the final results to a timestamped JSON file.
+"""
+
+from celery import Celery
+from celery import group
+import json
+import random
+from datetime import datetime
+from typing import List, Dict
+
+# Initialize Celery app
+app = Celery('etl_example')
+app.conf.update(
+    broker_url='redis://localhost:6379/0',
+    result_backend='redis://localhost:6379/0'
+)
+
+# Simulated data sources
+DATA_SOURCES = ['source_a', 'source_b', 'source_c']
+
+@app.task(name='etl.extract')
+def extract(source: str) -> List[Dict]:
+    """
+    Extract data from a simulated source.
+    In a real application, this could be a database, API, or file.
+    """
+    # Simulate fetching data with random delay
+    import time
+    time.sleep(random.uniform(0.1, 0.5))
+    
+    # Generate sample data
+    num_records = random.randint(5, 10)
+    records = []
+    
+    for i in range(num_records):
+        record = {
+            'id': f'{source}_{i}',
+            'timestamp': datetime.now().isoformat(),
+            'value': random.randint(1, 100),
+            'source': source
+        }
+        records.append(record)
+    
+    print(f'Extracted {len(records)} records from {source}')
+    return records
```

Comment:
I wonder - if I did this myself I might do this async and yield batches with a max size.